### DISCUSSION TITLE
Trouble with upgrading the Tethys Platform installed in the Azure virtual machine image

### DISCUSSION THREAD

**Discussion by ZoloKiala** [tethysplatform/tethys#1007] [Tutorial Help]
Dear All,

I'm trying to upgrade the Tethys Platform contained in the Azure virtual machine image. However, the upgrade doesn't work. The following command: conda install -c conda-forge -c tethysplatform tethys-platform=4.* doesn't work. I had to use the libmamba solver to make it work. Once fixed, when I tried "tethys db migrate", I got an error saying: the "ModuleNotFoundError: No module named 'tethys_portal'. I'm stuck. I can't use the Tethys version (3.3), which is in the image, because my app is built using version 4.2.

**Comment by romer8** [comment]
Hi @ZoloKiala, I do not if you can attach the content of your` portal_config.yml`. Similarly, I think the content of the following commands: `tethys version` and `tethys manage showmigrations` I think these contents can help me. In addition, the following logs will be helpful: `/var/log/tethys/error.log` 

I have updated Tethys from 3 to 4 before and some of the things that have worked for me are the following:

1 The python version of the current tethys env might be old to work with the package tethys-platform package. I think updating it to python 3.12 might help.

2  Install the new Tethys platform version with libmamba solver or with mamba (if you want to install tethys with mamba make sure mamba is installed in the base env, if you are using the libmamba solver you do not need to worry about this ) Also the version 4.2 is available only with the dev label

` mamba install -c conda-forge -c tethysplatform/label/dev tethys-platform=4.2`

or if you want to use the libmamba solver

`conda update -n base conda`
`conda install -n base conda-libmamba-solver`
`conda config --set solver libmamba`

3 Tethsy 4 versions need to specify the database in the portal_config.yml. You can do it with the following: 
`tethys settings --set DATABASES.default.ENGINE django.db.backends.postgresql --set DATABASES.default.NAME tethys_platform --set DATABASES.default.USER <TETHYS_DB_USERNAME> --set DATABASES.default.PASSWORD <TETHYS_DB_PASSWORD> --set DATABASES.default.HOST <TETHYS_DB_HOST> --set DATABASES.default.PORT <TETHYS_DB_PORT>`
Usually, you only need  `tethys settings --set DATABASES.default.ENGINE django.db.backends.postgresql`
Some Refs: https://docs.tethysplatform.org/en/stable/installation/production/manual/configuration/basic/database.html

4 Run `tethys db migrate`.

**Reply by ZoloKiala** [reply]
Thanks, @romer8 for the prompt reply. I followed the steps that you mentioned above. The Tethys version is 4.2. When I try to migrate the db, I'm getting the following error:
![image](https://github.com/tethysplatform/tethys/assets/65705483/e0cbd44d-19db-43df-93f7-e17c46fc295d)

**Reply by ZoloKiala** [reply]
Below is what I get with tethys manage showmigrations:
![image](https://github.com/tethysplatform/tethys/assets/65705483/e80384cf-4e1e-43fa-b003-973845829f3d)
![image](https://github.com/tethysplatform/tethys/assets/65705483/2d9b6cd4-bbd6-4b02-8ca0-8ac7b08aa937)

**Comment by ZoloKiala** [comment]
Below is the output of /var/log/tethys/error.log

![image](https://github.com/tethysplatform/tethys/assets/65705483/89678160-b3a4-41a3-9650-b5e7204753b8)

**Reply by romer8** [reply]
@ZoloKiala can you provide me with the content of the `portal_config.yml` thanks

**Reply by ZoloKiala** [reply]
@romer8 Thank you for your interest in replying to my query. I fixed the issue by uninstalling Postgresql and then, configuring it based on the instructions mentioned in the Production Installation Guide. However, I was not able to get the server up and running after migrating the tethys db. I'm getting a 502 Bad Gateway response.

Anyway, I decided to manually install the Tethys Platform and it was successful.  I noticed that, when I try to install the app, I'm not getting any error, but there is no app installed. See below the output on the terminal:

![image](https://github.com/tethysplatform/tethys/assets/65705483/8f9d8da5-6064-4ecd-8181-c38b7084fb73)

I'm trying now the installation of the app with Tethys-app store, but it's taking a lot of time. I guess it's the normal behaviour.

**Reply by romer8** [reply]
Hi @ZoloKiala Usually for a 502 Bad Gateway response, the response can be found in the logs at /var/log/tethys/error.log. I think it is possible that the app is failing at install. if you put the command tethys list, it should output the new app, but if not it means it was not installed and there is a possible error. What does the says  /var/log/tethys/error.log. or did it installed with the app_store?

**Reply by romer8** [reply]
@ZoloKiala I was able to reproduce why it is not installing manually. The problem is that if the permissions are given to the nginx-user at the moment you ran  `tethys install` it will not install the app. The folders where the static, workspaces , miniconda3, and the app soruce code need to be owned by the user.
Here some useful documentation: https://docs.tethysplatform.org/en/stable/installation/production/manual/configuration/basic/file_permissions.html

**Reply by romer8** [reply]
here soem documentation how to install apps: https://docs.tethysplatform.org/en/stable/installation/production/manual/app_installation.html#install-the-app

**Reply by ZoloKiala** [reply]
@romer8 I tried all the above suggestions, but I can't still install the app. No error is printed out. The Tethys app store is taking forever. Do you mind if you can have a video call to go through it together?

**Reply by romer8** [reply]
sure we can go through it together My time is USA Mountain Time. which time will it work better for you @ZoloKiala
I have a meeting link that you can use to schedule a meeting:  https://cal.com/giovanni-romero
also feel free to email me at gromero@aquaveo.com
